fastestimator/op/tensorop/augmentation/augmentation_3d.py

```diff
@@ -0,0 +1,163 @@
+                                                                                      0], [0, 0, 1, 0, 1,
+                                                                                           0], [0, 0, 1, 1, 0,
+                                                                                                0], [1, 1, 1, 0, 0, 0],
+                                                       [1, 1, 1, 1, 1,
+                                                        0], [0, 0, 1, 0, 0,
+                                                             0], [0, 0, 1, 1, 1,
+                                                                  0], [0, 1, 0, 1, 1,
+                                                                       1], [1, 1, 0, 0, 1,
+                                                                            1], [0, 0, 0, 1, 0,
+                                                                                 1], [1, 1, 0, 1, 0,
+                                                                                      1], [0, 1, 0, 0, 0, 1]])
+
+    # equivalent of  np.rot90(data, rotate_y, axes=(0, 2))
+    def _rotate_y(self, data):
+        data = tf.reverse(data, axis=[2])
+        data = tf.transpose(data, perm=[2, 1, 0])
+        return data
+
+    def _no_rotate(self, data):
+        return data
+
+    #equivalent of  np.rot90(data, rotate_z, axes=(0, 1))
+    def _rotate_z(self, data):
+        data = tf.reverse(data, axis=[1])
+        data = tf.transpose(data, perm=[1, 0, 2])
+        return data
+
+    def _flip_x(self, data):
+        data = tf.reverse(data, axis=[0])
+        return data
+
+    def _flip_y(self, data):
+        data = tf.reverse(data, axis=[1])
+        return data
+
+    def _flip_z(self, data):
+        data = tf.reverse(data, axis=[2])
+        return data
+
+    def _no_flip(self, data):
+        return data
+
+    def _transpose(self, data):
+        data = tf.transpose(data)
+        return data
+
+    def _no_transpose(self, data):
+        return data
+
+    def forward(self, data, state):
+        """Transforms the data with the augmentation transformation
+
+        Args:
+            data: Data to be transformed
+            state: Information about the current execution context
+
+        Returns:
+            Transformed (augmented) data
+
+        """
+        data, mask = data
+        key = tf.random.uniform(shape=(), minval=0, maxval=48, dtype=tf.dtypes.int32)
+        augment_key = tf.gather_nd(self.isometric_options, [key])
+        rotate_y = tf.cast(tf.random.uniform(shape=(1, ), minval=0, maxval=2, dtype=tf.dtypes.int32), tf.bool)
+        rotate_y = tf.cast(augment_key[0], tf.bool)
+        rotate_z = tf.cast(augment_key[1], tf.bool)
+        flip_x = tf.cast(augment_key[2], tf.bool)
+        flip_y = tf.cast(augment_key[3], tf.bool)
+        flip_z = tf.cast(augment_key[4], tf.bool)
+        transpose = tf.cast(augment_key[5], tf.bool)
+
+        data = tf.cond(rotate_y,
```

Comment:
is there a particular reason for tf.cond here instead of if/else auto-graphing?

Reply:
Modified the logic to the more readable  if/else style.